File: core/templates/test/java/{{packagePath}}/{{className}}Test.java
package {{package}};

import static org.junit.Assert.*;
import org.junit.Test;
import org.junit.Before;
import org.junit.After;

/**
 * Unit tests for {{className}} class.
 */
public class {{className}}Test {
    
    private {{className}} extension;
    
    @Before
    public void setUp() {
        // Note: In real tests, you would mock ComponentContainer
        extension = new {{className}}(null);
    }
    
    @After
    public void tearDown() {
        extension = null;
    }
    
    @Test
    public void testBasicFunctionality() {
        assertNotNull("Extension should be created", extension);
    }
    
    // Add more specific tests based on the extension type
}

File: core/templates/test/kotlin/{{packagePath}}/{{className}}Test.kt
package {{package}}

import org.junit.Test
import org.junit.Before
import org.junit.After
import kotlin.test.*

/**
 * Unit tests for {{className}} class.
 */
class {{className}}Test {
    
    private lateinit var extension: {{className}}
    
    @Before
    fun setUp() {
        // Note: In real tests, you would mock ComponentContainer
        extension = {{className}}(null)
    }
    
    @After
    fun tearDown() {
        // Cleanup if needed
    }
    
    @Test
    fun testBasicFunctionality() {
        assertNotNull("Extension should be created", extension)
    }
    
    // Add more specific tests based on the extension type
}